Sources/core/objects/Projectiles.cpp

```diff
@@ -0,0 +1,254 @@
+#include "Projectiles.h"
+#include "../Helpers.h"
+
+
+Projectile::Projectile(size_t id) : IMovable(id)
+{
+    SetPassable(D_ALL, Passable::AIR);
+    tickSpeed = 0.25;
+    damage = 0;
+    SetSprite("icons/projectiles.dmi");
+    v_level = 5;
+}
+void Projectile::Process()
+{
+    if(set_target_ == 0)
+    {
+       return;
+    }
+    int z = current_step_;
+    Dir step = VDirToDir(movement_[z]);
+
+   if(!TryMove(step))
+   {
+       this->Delete();
+       return;
+   }
+   current_step_++;
+   if(!(current_step_ < movement_.size()) && !reached_target )
+   {
+       current_step_ = 0;
+       VDir temp = movement_[0];
+       movement_.clear();
+       reached_target = true;
+       movement_.push_back(temp);
+   }
+   if(!(current_step_<movement_.size()) && reached_target )
+   {
+   current_step_ = 0;
+   }   
+}
+
+bool Projectile::CheckPassable()
+{
+    id_ptr_on<Projectile> p = this->GetId();
```

Comment:
I added one more varible which is name isleaser so ` if (!CanPass(owner->GetPassable(dMove), passable_level))
+    {
+        owner->Bump(p);
+        if (loc != Passable::FULL)
+        {
+            SetPassable(dMove, loc);
+        }
+        return false;`

This is what check whether there is a wall/window or anything, so I should change this if that's thing is a window then make it able to pass through it, should I overload the bump whit one more argument(so I can give islease variable to it and then to the tile) or I should search for an alternative way?